pkg/cinderbackup/statefulset.go

```diff
@@ -96,6 +98,12 @@ func StatefulSet(
 		volumeMounts = append(volumeMounts, instance.Spec.TLS.CreateVolumeMounts(nil)...)
 	}
 
+	// add MTLS cert if defined
+	if memcached.Status.MTLSCert != "" {
```

Comment:
as per the comments I had, if `MemcachedInstance` is a *string in the sub resources (for minor updates backward compatibility), this condition should also depends on memcached (`if memcached.Status.MTLSCert != "" && instance.Spec.MemcachedInstance != nil`)